daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/VariableMap1.scala

```diff
@@ -65,35 +82,26 @@ case object VariableInProcess extends VariableState
 
 /**
  * Class for maintaining the state of a variable
- *
- * Note: stateArg, valueArg, and defaultValueExprArg are pass by name/lazy
- * values as it is necessary to postpone their evaluation beyond when the
- * VariableInstance is created. Attempting to evaluate these arguments will
- * likely trigger an expression to be evaluated, which will query the
- * VariableMap if the expression references another variable. If this is
- * occurring within a defineVariable call, the VariableMap hasn't been fully
- * created and attempting to evaluate such an expression will result in an OOLAG
- * Circular Definition Exception
  */
 object VariableInstance {
   def apply(
-    stateArg: => VariableState,
-    valueArg: => DataValuePrimitiveNullable,
+    state: VariableState = VariableUndefined,
+    value: DataValuePrimitiveNullable = DataValue.NoValue,
     rd: VariableRuntimeData,
     defaultValueExprArg: => Maybe[CompiledExpression[AnyRef]],
     priorState: VariableState = VariableUndefined,
     priorValue: DataValuePrimitiveNullable = DataValue.NoValue) = {
 
-    new VariableInstance(stateArg, valueArg, rd, defaultValueExprArg, priorState, priorValue)
+    new VariableInstance(state, value, rd, defaultValueExprArg, priorState, priorValue)
   }
 }
 
 /**
  * See documentation for object VariableInstance
  */
 class VariableInstance private (
-  @TransientParam stateArg: => VariableState,
-  @TransientParam valueArg: => DataValuePrimitiveNullable,
+  var state: VariableState,
+  var value: DataValuePrimitiveNullable,
   val rd: VariableRuntimeData,
   @TransientParam defaultValueExprArg: => Maybe[CompiledExpression[AnyRef]],
```

Comment:
Thoughts on moving state, value, priorState, and priorValue to be members of this class, rather than constructor parameters? I think everywhere we create a VariableInstance now these values are always VairableUndefined/NoValue, and with your changes I can't think of a reason why we would ever need to create one of these with something different?

If so, and if we can remove defaultValueExpr, maybe the only parameter is the VariableRunimeData?